aiohttp/connector.py

```diff
@@ -805,11 +805,19 @@ def __init__(
         self._local_addr_infos = aiohappyeyeballs.addr_to_addr_infos(local_addr)
         self._happy_eyeballs_delay = happy_eyeballs_delay
         self._interleave = interleave
+        self._resolve_host_tasks: Set["asyncio.Task[List[ResolveResult]]"] = set()
 
-    def _close_immediately(self) -> List["asyncio.Future[None]"]:
+    def _close_immediately(self) -> List[Awaitable[object]]:
         for ev in self._throttle_dns_events.values():
             ev.cancel()
-        return super()._close_immediately()
+
+        waiters = super()._close_immediately()
+
+        for t in self._resolve_host_tasks:
+            t.cancel()
```

Comment:
Note to self: we have a shield below, verify that cancellation here can’t cause a consistency problem with the cache when I’m at my desktop

Reply:
cache is per connector so there is no concern about cache consistency when the connector is being closed down anyways.